presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -421,6 +421,19 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     }
 
     private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
+    {
+        try {
+            return doGetTableMetadata(tableName);
+        }
+        catch (TableNotFoundException | HiveViewNotSupportedException e) {
```

Comment:
Shouldn't we rethrow any kind of PrestoException? Maybe we could create a new `PrestoException` with the same message and error code, but also with information of affected table and rethrow?

Reply:
You're right.

Reply:
You're right, we should rethrow any `PrestoException` so that we can add better error handling downstream. 
I don't think modifying err msg of a `PrestoException` is a good idea. Their messages look more like sentences, so it's not easy to "inject" more information this way. I think the place where `PrestoException` originates needs to have all required information at hand.

Reply:
> Their messages look more like sentences, so it's not easy to "inject" more information this way.

I see following options: 
 - adding another sentence like `Something is not right with your metadata. Affects on tpch.nation table.`
 - or adding tags like `Something is not right with your metadata. [Affected table: tpch.nation]`?

Reply:
> I think the place where `PrestoException` originates needs to have all required information at hand.

True. Let's keep it this way.